### DISCUSSION TITLE
Help configuring fail2ban, cannot get IP from gogs log

### DISCUSSION THREAD

**Discussion by Dark-Schneider-666** [gogs/gogs#7841] [Q&A]
I use the official gogs/gogs docker image.

Version is:
```
d20d4400e629:/app/gogs# ./gogs --version
Gogs version 0.14.0+dev
```

The app.ini log part is:
```
[log]
MODE      = file
LEVEL     = Info
ROOT_PATH = /app/gogs/log
```
The gogs.log file has not authentication info, can only get from gorm.log, but it only logs this:
```
d20d4400e629:/app/gogs/log# cat gorm.log
2024/11/17 09:00:20 gogs.io/gogs/internal/database/users.go:80 record not found
[0.503ms] [rows:0] SELECT * FROM `user` WHERE lower_name = "test" ORDER BY `user`.`id` LIMIT 1
2024/11/17 09:14:30 gogs.io/gogs/internal/database/migrations/migrations.go:82 SLOW SQL >= 100ms
[116.145ms] [rows:1] SELECT * FROM `version` WHERE id = 1 ORDER BY `version`.`id` LIMIT 1
```
So I can get the user ID but not the IP.

Need some help for loggin IP address from authentication attemps. Thanks.

**Comment by derak-kilgo** [comment]
For the web traffic, put a proxy around it. 
The docs have several examples using caddy, nginx or apache.
Plenty of examples of getting those hooked up with fail2ban.

To get the ssh logins, you'd need to adjust your docker setup a bit. The docker image uses sshd inside the container. You'd need to expose that log so fail2ban can watch it.

https://github.com/gogs/gogs/tree/main/docker
sshd_conig is one of the files the docker container customizes. 
You could change where the logs output. (say to the /data/gogs/log folder)

I'll say I dug around in the container and couldn't find any logs. I expected them to be in /var/log and even found log rotate scripts which pointed there but I didn't see any actual log files.
